cpp/src/parquet/encryption/file_key_unwrapper.cc

```diff
@@ -27,14 +27,22 @@ namespace encryption {
 
 using internal::KeyWithMasterId;
 
-FileKeyUnwrapper::FileKeyUnwrapper(KeyToolkit* key_toolkit,
-                                   const KmsConnectionConfig& kms_connection_config,
-                                   double cache_lifetime_seconds)
+FileKeyUnwrapper::FileKeyUnwrapper(
+    KeyToolkit* key_toolkit, const KmsConnectionConfig& kms_connection_config,
+    double cache_lifetime_seconds, const std::shared_ptr<FilePath>& parquet_file_path,
+    std::shared_ptr<FileKeyMaterialStore> key_material_store)
     : key_toolkit_(key_toolkit),
       kms_connection_config_(kms_connection_config),
-      cache_entry_lifetime_seconds_(cache_lifetime_seconds) {
+      cache_entry_lifetime_seconds_(cache_lifetime_seconds),
+      checked_key_material_internal_storage_(false),
+      parquet_file_path_(parquet_file_path) {
   kek_per_kek_id_ = key_toolkit_->kek_read_cache_per_token().GetOrCreateInternalCache(
       kms_connection_config.key_access_token(), cache_entry_lifetime_seconds_);
+
+  if (key_material_store != nullptr) {
+    key_material_store_ = key_material_store;
+    checked_key_material_internal_storage_ = true;
```

Comment:
What if instead of having this bool we just checked `key_material_store != nullptr`?